### PR TITLE
Remove originalexpression from optimizer

### PR DESCRIPTION
**Pull request by pranjalssh**
This PR removes usage of OriginalExpression from all optimizer rules. The first rule: TranslateExpressions translates all OriginalExpression -> proper RowExpression.

This PR will unblock crux. We can stop here, or go further and remove OriginalExpression usage from planner and tests as a code cleanup

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java [hunk 1]
```diff
@@ -315,11 +314,6 @@ public PlanOptimizers(
                                 .addAll(new SimplifyCardinalityMap().rules())
                                 .addAll(new DesugarRowSubscript(metadata, sqlParser).rules())
                                 .build()),
-                new IterativeOptimizer(
```
Comment (jainxrohit) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java:318]: Why is this removed? Can you add detail in the commit?

Comment (pranjalssh) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java:318]: This logic is moved to TranslateExpressions

**Comment by jainxrohit** [comment]
I also tested this with PRUX  in the verifier cluster. It works. Lets add PR description.

**Review by jainxrohit — APPROVED** [review]
